compiler/src/iree/compiler/Dialect/Util/Transforms/Patterns.cpp

```diff
@@ -496,6 +571,308 @@ struct MergeIndexSwitchPattern : public OpRewritePattern<scf::IndexSwitchOp> {
   }
 };
 
+// Simplifies scf.if when one branch contains only util.unreachable.
+//
+// Example:
+//  scf.if %cond {
+//    util.unreachable
+//  } else {
+//    "some.op"() : () -> ()
+//  }
+// ->
+//  // TODO: add assertion that !%cond when we have util.assume
```

Comment:
when we have `util.assert`? Why we prefer assume over assert? I think we have `cf.assert`, maybe we can use that?

Reply:
I see, thanks for the details!